### PR TITLE
NIFI-9645 - Updated PutSplunk to enable idle connection timeouts.

### PR DESCRIPTION
**Pull request by thenatog**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by exceptionfactory — COMMENTED** [review]
Thanks for this improvement @thenatog.  Introducing the idle timeout makes sense, and the implementation looks straightforward.

The only notable issue is changing the existing `timeout` to `connectionTimeout`. In the current implementation, `timeout` applies to both the initial connection, as well as subsequent write operations.  One option is to leave the existing property name as `timeout`, and just introduce the new `idleTimeout`.  The other option would be to create both `connectionTimeout` and `writeTimeout` as separate properties.  Given the addition of `idleTimeout`, the second option seems like the better way to go. What do you think?

File: nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSenderFactory.java [hunk 1]
```diff
@@ -61,6 +61,8 @@
 
     private Duration timeout = Duration.ofSeconds(30);
 
+    private Duration idleTimeout = Duration.ofSeconds(30);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSenderFactory.java:64]: It seems like the internal default should be `0` given that no idle timeout is the current internal behavior of this class.

File: nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/channel/StandardChannelInitializer.java [hunk 2]
```diff
@@ -37,6 +39,8 @@
 
     private Duration writeTimeout = Duration.ofSeconds(30);
 
+    private Duration idleTimeout = Duration.ofSeconds(30);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/channel/StandardChannelInitializer.java:42]: As in the `NettyEventSenderFactory`, it seems like the internal default should be `0`.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for updating this PR @thenatog. The current version looks close to completion and the idle timeout appears to behave as expected at runtime.

One final note is the internal default value of the idle timeout in the handler and factory classes.  The default value in the processor property looks good, but for the Netty-based classes, it seems like the internal default should be `0` so that idle timeout is disabled unless specified. What do you think?

File: nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSenderFactory.java (hunk 1 above)
Comment (thenatog) [reply nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSenderFactory.java:64]: Sure that makes sense, 30 seconds did seem arbitrary given the idle timeout default is now 15 seconds.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @thenatog, this looks good! +1 merging.

**Pull request closed without merging**